### ISSUE TITLE
unnecessary return

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#2836]
**Description**
Some `return` statements are redundant, and can safely be removed, if they are not followed by other statements.

This is specially true for `void` functions.

I am not sure if a lint exists, or should it be part of [unnecessary_statements](https://dart-lang.github.io/linter/lints/unnecessary_statements.html)?

**Examples**

```dart
void m(bool x) {
  if (x) {
    doSomething();
    return; // LINT
  }
  return; // LINT
}

void m2(bool x) {
  if (x) {
    doSomething();
    return; // OK
  }
  another();
  return; // LINT
}

```

**Comment by bwilkerson** [comment]
> I am not sure if a lint exists, or should it be part of unnecessary_statements?

I don't believe there is currently such a lint. Do you have an estimate for how often this occurs? I personally don't remember ever seeing it in a code review, but that doesn't prove anything.

I believe that the primary purpose of 'unnecessary_statements' is to find places that are likely to be bugs because of missing code. This kind of error doesn't really qualify as a bug. But if we think it's worth adding support for it we might consider adding it to the 'dead_code' support.

**Comment by asashour** [comment]
Actually, two unneeded `return`s are currently in `master` and were detected (and removed) in #2800.

https://github.com/dart-lang/linter/blob/def6bf20f22a8b974673606a942f392700efef31/lib/src/rules/unnecessary_parenthesis.dart#L127-L131
